### PR TITLE
[WIP] Dev guide

### PR DESCRIPTION
**Pull request by Bradamant3**
Current state: content reorg only begun. Not sure it's quite right. Questions for reviewers:

- there's a sort of gray area between using and extending Ark, especially w/ hooks. Not sure where this should go.
- ditto plug-ins, although they seem a bit more obviously to belong to a dev guide (or a pointer to the repo, at least)

### CODE REVIEW

File: docs/developing/dev-prereq.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  * Write access to the target cluster and namespaces
+* Cloud provider credentials
+  * Read/write access to volumes
+  * Read/write access to object storage for backup data
+* A [BackupStorageLocation][20] object definition for the Ark server
+* (Optional) A [VolumeSnapshotLocation][21] object definition for the Ark server, to take PV snapshots
+
+For detailed examples, see the YAML files in the [examples directory][6].
+
+TODO FROM JFR: how much introductory/explanatory material should be provided here? This is an audience question. To date, it seems as though outside contributors to Ark have been primarily folks deeply familiar with K8s and the dev environment it requires. PR #1131 got me started wondering about k8s newcomers -- although that user seems not to have been a potential code contributor (at least, not yet). IFF y'all want to support newcomer contributions, I think there's quite a bit of material missing. See also TODOs in other files.
```
Comment (skriss) [inline docs/developing/dev-prereq.md]: Add some refs to CRDs/controllers as relevant pre-req knowledge

File: docs/developing/dev-test.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+* Test your build
```
Comment (skriss) [inline docs/developing/dev-test.md]: Add some general guidelines on getting your build deployed, some standard flows to go through, where to look for success/failure/diagnostic info

**Comment by Bradamant3** [comment]
n.b. please long TODO comment in new `documentation.md`. More food for thought, approaches, ideas, reminders of additional content to include.

**Comment by skriss** [comment]
Sorry @Bradamant3 but I'm going to close this PR as it's stale :) We have xrefs to it in the relevant issues so we can mine it for content if/when we work on those issues.

**Pull request closed without merging**